src/agent/service/nodesCrossSignClaim.ts

```diff
@@ -65,10 +68,23 @@ function nodesCrossSignClaim({
         if (payloadData.type !== 'node') {
           throw new claimsErrors.ErrorNodesClaimType();
         }
-        // Verify the claim
-        const senderPublicKey = await nodeManager.getPublicKey(
-          nodesUtils.decodeNodeId(payloadData.node1),
+        const { nodeId }: {
+          nodeId: NodeId,
+        } = await validateSync(
+          (keyPath, value) => {
+            return matchSync(keyPath)(
+              [
+                ['nodeId'],
+                () => validationUtils.parseNodeId(value)
+              ],
+            );
+          },
+          {
+            nodeId: payloadData.node1,
+          }
```

Comment:
This seems to be causing problems for `can successfully cross sign a claim` in `tests/agent/GRPCClientAgent.test.ts` - I'm getting `ErrorPolykey: Cannot destructure property 'nodeId' of '(intermediate value)' as it is undefined.` when including this. Reverting it to the original `decodeNodeId` call (without any validation) causes it to continue with the test (however, there's another problem down the line with the test too - see next comment from me).

Reply:
Doesn't this mean that the test is broken?

As it is done now, it should be validating that the node id is in fact a node id.

Also wrong usage of `await` here, I'm removing it.

Reply:
The reason for this is the lack of the default case of `() => value`. I'm adding this in now.